src/Api/Vault/AuthorizationHandlers/OrganizationUsers/OrganizationUserAuthorizationHandler.cs

```diff
@@ -76,10 +79,33 @@ public class OrganizationUserAuthorizationHandler : AuthorizationHandler<Organiz
             return;
         }
 
+        // Check for non-null org here: the user must be apart of the organization for this setting to take affect
+        // If the limit collection management setting is disabled, allow any user to read all organization users
+        if (org is not null && await GetOrganizationAbilityAsync(org) is { LimitCollectionCreationDeletion: false })
```

Comment:
@eliykat I think we might have missed removing this extra `org is not null` check